### PR TITLE
{bio}[gompi/2021b] MEME v5.4.1 w/ Python 2.7.18

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

requires:
* #15176
* #15177

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node072.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/d123446f9dff7e63859155f9c242ad95 for a full test report.

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+version = '5.4.1'
+
+homepage = 'https://meme-suite.org/meme/index.html'
+description = """The MEME Suite allows you to: * discover motifs using MEME, DREME (DNA only) or
+ GLAM2 on groups of related DNA or protein sequences, * search sequence databases with motifs using
+ MAST, FIMO, MCAST or GLAM2SCAN, * compare a motif to all motifs in a database of motifs, * associate
+ motifs with Gene Ontology terms via their putative target genes, and * analyse motif enrichment
+ using SpaMo or CentriMo."""
+
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
```
Comment (jfgrimm) [inline easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb:17]: `MEME` claims to have `MPI` support, is there a particular reason for building without it?

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+dependencies = [
+    ('libxml2', '2.9.10'),
+    ('libxslt', '1.1.34'),
+    ('zlib', '1.2.11'),
+    ('Perl', '5.34.0'),
+    ('Python', '3.9.6')
+]
+
+configopts = '--enable-build-libxml2 --enable-build-libxslt '
```
Comment (jfgrimm) [inline easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb:31]: according to the docs, this builds the bundled versions of `libxml2`, `libxslt`. These flags should be omitted if you want to actually link the dependencies EB provides

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+configopts += '--with-perl=${EBROOTPERL}/bin/perl --with-python3=${EBROOTPYTHON}/bin/python '
+
+sanity_check_paths = {
+    'files': ['bin/dreme', 'bin/meme', 'bin/dreme', 'bin/meme-chip'],
+    'dirs': ['lib'],
+}
+
```
Comment (jfgrimm) [inline easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb:38]: should probably add `libexec` to `PATH`, and probably ensure the perl modules are properly picked up as well
```suggestion

modextrapaths = {
    'PATH': ['libexec'],
    'PERL5LIB': ['lib/%(namelower)s-%(version)s/perl'],
}

```

**Review by jfgrimm — CHANGES_REQUESTED** [review]
running the provided `scripts/dependencies.pl` script, we're missing some Perl modules as well:
`$PERL scripts/dependencies.pl

Checking Required Perl Modules:
HTML::Template missing.
Sys::Info missing.

Checking Optional Modules:
XML::Compile::SOAP11 missing. Used for downloading sequence databases from RSAT.
On Ubuntu, you may first need to run 'sudo apt-get install libxml2-dev'
XML::Compile::WSDL11 missing. Used for downloading sequence databases from RSAT
XML::Compile::Transport::SOAPHTTP missing. Used for downloading sequence databases from RSAT`

**Comment by jfgrimm** [comment]
python shebangs are also messed up for some of the scripts, looks like the substitution rules in `scripts/Makefile.am` are broken:
```
$ head -n 1 $EBROOTMEME/lib/meme-5.4.1/python/hypergeometric_py3.py
#!@WHICHPYTHON@
```

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb:31]: Ouch, good catch.

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb:17]: Hmm, no particular reason, I could look into building `meme` with MPI support instead, which probably boils down to using `gompi` as toolchain than `GCC`...

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb (hunk 3 above)
Comment (boegel) [reply easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb]: fixed in [752d645](https://github.com/easybuilders/easybuild-easyconfigs/pull/15169/commits/752d6454b6437d018cc0c59b1224c59690538985)

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb]: fixed in 752d645

File: easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/m/MEME/MEME-5.4.1-GCC-11.2.0.eb]: fixed in 752d645

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3110.skitty.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/1d5916a4bd5072f6792277bdcc5b8f4d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3531.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/8855bdbccfde7f18db3e88fba81e9566 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in total)
node138.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/bb6719b242e7f7e38f9d1e600d8fb53a for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login2.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/4770ca9411fe0c376591395f6c82eaa1 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
login2.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/3e6e5561deb9aa6e2bdbb21119085fb8 for a full test report.

**Comment by jfgrimm** [comment]
I tried adding `runtest = test` and it failed with:
```11285 Operating system identified as: 'Linux'. Native driver can not be loaded: Error loading Sys::Info::Driver::Linux::Device::CPU: Can't locate Sys/Info/Driver/Linux/Device/CPU.pm in @INC       (you may need to install the Sys::Info::Driver::Linux::Device::CPU module) (@INC contains: /tmp/jfg508/MEME/5.4.1/gompi-2021b-Python-2.7.18/meme-5.4.1/scripts /users/jfg508/scratch/eas      ybuild-dev/software/XML-Compile/1.63-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/XML-Compile/1.63-GCCcore-11.2.0/li      b/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/XML-LibXML/2.0207-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybu      ild-dev/software/XML-LibXML/2.0207-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-l      inux-thread-multi /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11      .2.0/lib/perl5/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/5.34.0) at /users/jfg508/scratch/easybuild-dev/softwar      e/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Base.pm line 45.
11286  at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Device/CPU.pm line 9.
11287 . Falling back to compatibility mode                                                                                    
11288 Error loading Sys::Info::Device::CPU: Error loading Sys::Info::Driver::Unknown::Device::CPU: Can't locate Sys/Info/Driver/Unknown/Device/CPU.pm in @INC (you may need to install the Sys      ::Info::Driver::Unknown::Device::CPU module) (@INC contains: /tmp/jfg508/MEME/5.4.1/gompi-2021b-Python-2.7.18/meme-5.4.1/scripts /users/jfg508/scratch/easybuild-dev/software/XML-Compil      e/1.63-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/XML-Compile/1.63-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /user      s/jfg508/scratch/easybuild-dev/software/XML-LibXML/2.0207-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/XML-LibXML/2.      0207-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg50      8/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/5.34.0/x86_64-l      inux-thread-multi /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/5.34.0) at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/l      ib/perl5/site_perl/5.34.0/Sys/Info/Base.pm line 45.
11289  at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Device/CPU.pm line 9.
11290 BEGIN failed--compilation aborted at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Device/CPU.pm line 9.
11291 Compilation failed in require at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Base.pm line 45.
11292  at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info.pm line 79.
```

doesn't seem to affect actual program execution though, running on some of the example inputs I had no issues

**Comment by boegel** [comment]
> I tried adding `runtest = 'test'` and it failed with:
> 
> ```
> 11286  at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Device/CPU.pm line 9.
> 11287 . Falling back to compatibility mode                                                                                    
> 11288 Error loading Sys::Info::Device::CPU: Error loading Sys::Info::Driver::Unknown::Device::CPU: Can't locate Sys/Info/Driver/Unknown/Device/CPU.pm in @INC (you may need to install the Sys::Info::Driver::Unknown::Device::CPU module) (@INC contains: /tmp/jfg508/MEME/5.4.1/gompi-2021b-Python-2.7.18/meme-5.4.1/scripts /users/jfg508/scratch/easybuild-dev/software/XML-Compile/1.63-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/XML-Compile/1.63-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/XML-LibXML/2.0207-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/XML-LibXML/2.0207-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0 /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/5.34.0/x86_64-linux-thread-multi /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/5.34.0) at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Base.pm line 45.
> 11289  at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Device/CPU.pm line 9.
> 11290 BEGIN failed--compilation aborted at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Device/CPU.pm line 9.
> 11291 Compilation failed in require at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info/Base.pm line 45.
> 11292  at /users/jfg508/scratch/easybuild-dev/software/Perl/5.34.0-GCCcore-11.2.0/lib/perl5/site_perl/5.34.0/Sys/Info.pm line 79.
> ```
> 
> doesn't seem to affect actual program execution though, running on some of the example inputs I had no issues

@jfgrimm fixed with #15190, MEME tests enabled in #15191

**Pull request merged**